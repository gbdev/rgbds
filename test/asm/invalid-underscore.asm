; good
println 123_456
println %_1010_1010
println 0b_1010_1010
println &_555_555
println 0o_777_777
println $_dead_beef
println 0x_cafe_babe
println `_0101_2323
println 12_34.56_78
println 12_34.56_q8

; bad (multiple '_')
println 123__456
println %1010__1010
println &123__456
println $abc__def
println `0101__2323
println 3.14__15
println 2.718__Q16

; bad (trailing '_')
println 12345_
println 0b101010_
println 0o123456_
println 0xabcdef_
println `01230123_

; bad ('_' next to '.')
println 1_.618
println 2._718
