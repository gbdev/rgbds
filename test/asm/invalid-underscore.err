error: Invalid integer constant, '_' after another '_'
    at invalid-underscore.asm(14)
error: Invalid integer constant, '_' after another '_'
    at invalid-underscore.asm(15)
error: Invalid integer constant, '_' after another '_'
    at invalid-underscore.asm(16)
error: Invalid integer constant, '_' after another '_'
    at invalid-underscore.asm(17)
error: Invalid integer constant, '_' after another '_'
    at invalid-underscore.asm(18)
error: Invalid integer constant, '_' after another '_'
    at invalid-underscore.asm(19)
error: Invalid integer constant, '_' after another '_'
    at invalid-underscore.asm(20)
error: Invalid integer constant, trailing '_'
    at invalid-underscore.asm(23)
error: Invalid integer constant, trailing '_'
    at invalid-underscore.asm(24)
error: Invalid integer constant, trailing '_'
    at invalid-underscore.asm(25)
error: Invalid integer constant, trailing '_'
    at invalid-underscore.asm(26)
error: Invalid graphics constant, trailing '_'
    at invalid-underscore.asm(27)
error: Invalid integer constant, trailing '_'
    at invalid-underscore.asm(30)
error: Invalid integer constant, '_' after another '_'
    at invalid-underscore.asm(31)
Assembly aborted with 14 errors!
