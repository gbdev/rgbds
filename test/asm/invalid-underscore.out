$1E240
$AA
$AA
$2DB6D
$3FFFF
$DEADBEEF
$CAFEBABE
$F55
$4D2915B
$4D28F
$1E240
$AA
$A72E
$ABCDEF
$F55
$32439
$2B7CF
$3039
$2A
$A72E
$ABCDEF
$3355
$19E35
$2B7CF
